vehicle/mkvii/software/brakes/bspd.c

```diff
@@ -0,0 +1,124 @@
+#include "bspd.h"
+
+#include "libs/adc/api.h"
+#include "libs/gpio/api.h"
+#include "libs/timer/api.h"
+#include "vehicle/mkvi/software/brakes/can_api.h"
+
+#include <avr/interrupt.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "projects/btldr/btldr_lib.h"
+#include "projects/btldr/git_sha.h"
+#include "projects/btldr/libs/image/api.h"
+
+// Required for btldr
+image_hdr_t image_hdr __attribute__((section(".image_hdr"))) = {
+    .image_magic = IMAGE_MAGIC,
+    .git_sha = STABLE_GIT_COMMIT,
+};
+
+// CAN interrupt + callback function
+volatile bool send_can = false;
+void timer0_callback(void) {
+    send_can = true;
+}
+
+// Code to run when a digital sense pin changes state (High -> Low or Low -> High)
+void pcint0_callback(void) {
+    // Update CAN struct with new board logic values
+    bspd.brake_gate = !!gpio_get_pin(BRAKELIGHT_LL);
+    bspd.bspd_5kw = !!gpio_get_pin(MOTOR_CURRENT_SENSE);
+    bspd.ss_bspd = !gpio_get_pin(BSPD_LL);
+}
+
+// Check whether an LED needs updating, and if so, change its state
+void update_LEDs(void) {
+    // Update Brake Light LED on the PCB
+    if (bspd.brake_gate == true) {
+        gpio_set_pin(BRAKE_LL_LED);
+    } else {
+        gpio_clear_pin(BRAKE_LL_LED);
+    }
+
+    // Update 5kW LED on the PCB
+    if (bspd.bspd_5kw) {
+        gpio_set_pin(MOTOR_5KW_LED);
+    } else {
+        gpio_clear_pin(MOTOR_5KW_LED);
+    }
+
+    // Update BSPD Status LED/BSPD Trip LED on the PCB
+    if (bspd.ss_bspd) {
+        gpio_set_pin(BSPD_TRIP_LED);
+    } else {
+        gpio_clear_pin(BSPD_TRIP_LED);
+    }
+
+    // Update Heartbeat LED on the PCB; Actions chosen to make LED blink every cycle (showing heartbeat)
+    if (bspd.heartbeat == true) {
```

Comment:
Please fix the heartbeat LED to blink at a human-readable interval using your 100hz CAN timer